test_package/conanfile.py

```diff
@@ -12,10 +12,13 @@ def build(self):
         cmake.build()
 
     def test(self):
-        with tools.environment_append(RunEnvironment(self).vars):
-            if self.settings.os == "Windows":
-                self.run(os.path.join("bin","test_package"))
-            elif self.settings.os == "Macos":
-                self.run("DYLD_LIBRARY_PATH=%s %s"%(os.environ.get('DYLD_LIBRARY_PATH', ''),os.path.join("bin","test_package")))
-            else:
-                self.run("LD_LIBRARY_PATH=%s %s"%(os.environ.get('LD_LIBRARY_PATH', ''),os.path.join("bin","test_package")))
+        if tools.cross_building(self.settings):
+            self.output.warn("Skipping run cross built package")
+        else:
+            with tools.environment_append(RunEnvironment(self).vars):
```

Comment:
Thanks for the tip. New convention is much cleaner; I had to hold my nose when making my previous edits, as it certainly didn't seem to be the right place to be open-coding OS-specific handling.

Reply:
Well, back then the Conan feature to handle this didn't exist and it only exists now because we noticed that we need this in pretty much all recipes and pushed it upstream😄